cfpb/consumerfinance.gov
# Include all answer_content text content in search index

Standardizing on the use of `get_searchable_content` so we don't have to name individual blocks, which isn't very future-proof. We were already not indexing HowTo and FAQ schema blocks, which was the main driver of this work.

## Changes

- Updates the `AnswerPage.text()` method that populates the ES7 index to call a single `answer_text()` method and prepend that with the `question` text
- `answer_text()` rolls up all of the functionality previously found in `short_answer_data()` and `answer_content_text()` (as well as `extract_raw_text()`, which `answer_content_text()` also called)
- Renamed `answer_content_data()` to `answer_content_preview()` to make its purpose more clear


## How to test this PR

1. Pull branch
1. `docker-compose up`
1. Load up an AnswerPage in the Wagtail admin and add a bunch of junk to its content StreamField, including something with a unique searchable term that won't already be in the DB, and including either a HowTo or FAQ schema block
1. `docker-compose exec python bash`
1. `./cfgov/manage.py search_index --rebuild -f --models ask_cfpb`
1. Open the Kibana console: http://localhost:5601/app/kibana#/dev_tools/console
1. Run this query, replacing `banana` with whatever unique term you added above:
   ```json
   GET /ask-cfpb/_search
   {
     "query": {
       "match": {
         "text": "narrator"
       }
     }
   }
   ```
1. Observe that the results return the page you edited, and that the `text` field includes the content you added in all blocks


## Notes and todos

- ~~I think the remaining uses of `extract_raw_text` can be removed, but I wanted to confirm with folks more familiar with the initial implementation. Will add a comment about this shortly.~~


## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
